src/Score.js

```diff
@@ -0,0 +1,142 @@
+function Game(frame = new Frame()) {
+  // This tracks the overall player score
+  this.score = 0
+  this.frames = [frame]
+  this.cumulatives = []
+
+  //THE BELOW ARE REFERENCED A LOT. MAYBE MAKE THEM INSTANCE VARIABLES
+  // this.lenFrames = this.frames.length
+  // this.currentFrame = this.frames[this.lenFrames - 1]
+  // this.previousFrame = null
+  // this.twobackframe = null
+  // this.lenCumulatives = this.cumulatives.length
+  // this.framePendingTotal = this.frames[lenCumulatives]
+
+};
+
+Game.prototype.add_score = function(score) {
+  currentframe = this.frames[this.frames.length - 1]
+  currentframe.add_score(score);
+
+  this._checkPrevBonusAndCumul()
+  // adds first pending bonus. Need to check again for if there is a second pending bonus.
+  this._checkPrevBonusAndCumul()
+
+  if (this._currentFrameOver()) {
+    this._checkCurrentFrameBonus()
+    this.score = this.cumulatives[this.cumulatives.length - 1]
+    this._add_frame();
+  }
+};
+
+Game.prototype._checkPrevBonusAndCumul = function () {
+  if (this._prevHasBonus()) {
+    if (this._bonusNotAdded()) {
+      if (this.allBonusBallsRolled()) {
+        this._add_bonus();
+      }
+    }
+  }
+};
+
+Game.prototype._checkCurrentFrameBonus = function() {
+  // if current frame not waiting for a bonus then add the cumuative score
+  lenFrames = this.frames.length
+  currentframe = this.frames[lenFrames - 1]
+  if (currentframe.bonusBalls === 0) {
+    this.cumulatives.push(this.newTotal())
+  }
+}
+
+Game.prototype._prevHasBonus = function () {
+  lenFrames = this.frames.length
+  currentframe = this.frames[lenFrames - 1]
+  if (lenFrames > 1) {
+    //doesn't check for bonus if on the first frame still
+    previousFrame = this.frames[lenFrames - 2]
+    if (previousFrame.bonusBalls > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
+
+Game.prototype._bonusNotAdded = function() {
+  lenFrames = this.frames.length
+  previousFrame = this.frames[lenFrames - 2]
+  if (previousFrame.bonusAdded === false) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+Game.prototype.allBonusBallsRolled = function () {
+    lenFrames = this.frames.length
+    lenCumulatives = this.cumulatives.length
```

Comment:
This logic confuses me. I'm not sure why you are maintaining cumulatives as independent state - you seem to be tending towards pushing as much stuff into state variables as possible when a lot of what you are holding there could be calculated dynamically. The cumulative score is just the sum of all the frame scores (with bonuses), so why not just have a method getOverallScore() which gets the sum of all the frame scores?